### PR TITLE
[SPARK-12807] [YARN] Spark External Shuffle not working in Hadoop clusters with Jackson 2.2.3: branch-1.6 patch

### PR DESCRIPTION
**Pull request by steveloughran**
This is the patch of PR #10780 applied to branch-1.6, to verify it works there too

### CODE REVIEW

File: network/yarn/pom.xml [hunk 1]
```diff
@@ -86,6 +88,15 @@
               </excludes>
             </filter>
           </filters>
+          <relocations>
```
Comment (srowen) [inline network/yarn/pom.xml]: How about doing this for all of Spark?

File: network/yarn/pom.xml [hunk 2]
```diff
@@ -96,6 +107,54 @@
           </execution>
         </executions>
       </plugin>
+
+      <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
```
Comment (srowen) [inline network/yarn/pom.xml]: This doesn't seem worth the complexity; why would the shading fail?

Comment (steveloughran) [inline network/yarn/pom.xml]: regression control. How do you think I got it to work in the first place? How to verify that it keeps working? I want to add some more shading; for that I'll use a small ant macro to make things terser. 

Consider this: A build is an execution of a program; programs needs tests. This is the test of the shading process.

File: network/yarn/pom.xml (hunk 1 above)
Comment (steveloughran) [inline network/yarn/pom.xml]: I'm now thinking all the dependencies should be relocated, with the exception of the jni leveldb stuff —not the thing you can shade; not the thing you can isolate in a classloader hierarchy instead.

What I can do there is add a profile, `leveldb-provided` which excludes leveldb in this situation. That's important as leveldb does creep onto the classpath from elsewhere —and while the version in Hadoop currently matches that in this JAR, there's no guarantee of that continuing to hold.

**Comment by steveloughran** [comment]
Sean: the PR #10780 is the version against /master; its the one with the full diff. This is just here to show it works for 1.6 too. How about I close this one; the other is used for review, and, when it's good, it can be pulled in to 1.6. So far the dependencies between 1.6 and master are identical as far as the shuffle goes.

**Comment by steveloughran** [comment]
(closing)

**Pull request closed without merging**